File: PTSA/code/ch02/smoothing_data.py
"""
@File         : smoothing_data.py
@Author(s)    : Stephen CUI
@LastEditor(s): Stephen CUI
@CreatedTime  : 2024-03-24 22:38:36
@Email        : cuixuanstephen@gmail.com
@Description  : 平滑数据
"""

import pandas as pd

air = pd.read_csv("../data/AirPassengers.csv", parse_dates=True, header=None)
air.columns = ["Date", "Passengers"]

air["Smooth.5"] = air["Passengers"].ewm(alpha=0.5).mean()
air["Smooth.1"] = air["Passengers"].ewm(alpha=0.1).mean()
air["Smooth.9"] = air["Passengers"].ewm(alpha=0.9).mean()

File: PTSA/code/ch02/smoothing_data.R
require(data.table)

air <- fread("../data/AirPassengers.csv")
names(air) <- c("Date", "Num")
air[, Date := as.Date(paste0(Date, "-01"), format = "%Y-%m-%d")]
loess.smooth(air$Date, air$Num)

plot(stl(AirPassengers, "periodic"))
